### PR TITLE
[NewTxType] for multiple currency chains

### PR DESCRIPTION
**Pull request by YunSuk-Yeo**
Multiple currency chains usually pay fee and send with multiple currency

* Append TxMultiCurrency & TxMultiCurrencyAnyAction type

### CODE REVIEW

File: pkg/blockatlas/tx.go [hunk 1]
```diff
@@ -21,13 +21,15 @@ const (
...
-	TxAnyAction           TransactionType = "any_action"
+	TxTransfer               TransactionType = "transfer"
+	TxMultiCurrencyTransfer  TransactionType = "multi_currency_transfer"
+	TxNativeTokenTransfer    TransactionType = "native_token_transfer"
+	TxTokenTransfer          TransactionType = "token_transfer"
+	TxCollectibleTransfer    TransactionType = "collectible_transfer"
+	TxTokenSwap              TransactionType = "token_swap"
+	TxContractCall           TransactionType = "contract_call"
+	TxAnyAction              TransactionType = "any_action"
+	TxMultiCurrencyAnyAction TransactionType = "multi_currency_any_action"
```
Comment (vikmeup) [inline pkg/blockatlas/tx.go]: Why calling `multi_currency_any_action` and not `multi_currency_transfer` ?

File: pkg/blockatlas/tx.go [hunk 2]
```diff
@@ -91,6 +93,8 @@ type Tx struct {
 	Meta interface{} `json:"metadata"`
 }
 
+// TxOutput describes input and output of tx
+// Excample: bitcoin UTXO
```
Comment (vikmeup) [inline pkg/blockatlas/tx.go]: `// Excample: bitcoin UTXO` why adding this? + misspelled

File: pkg/blockatlas/tx.go [hunk 3]
```diff
@@ -147,6 +151,30 @@ type ContractCall struct {
...
+	Value      Amount `json:"value"`
+	Symbol     string `json:"symbol"`
+	Decimals   uint   `json:"decimals"`
+	CurrencyID string `json:"currency_id"`
+}
+
+// MultiCurrencyTransfer describes the transfer of multiple currency native to the platform
+// Example: Cosmos, Terra
+type MultiCurrencyTransfer struct {
+	Currencies []Currency `json:"currencies"`
```
Comment (vikmeup) [inline pkg/blockatlas/tx.go]: you should use `blockatlas.Token`

File: pkg/blockatlas/tx.go (hunk 1 above)
Comment (YunSuk-Yeo) [reply pkg/blockatlas/tx.go]: I added two type of tx, `multi_currency_any_action` and `multi_currency_transfer`

File: pkg/blockatlas/tx.go (hunk 2 above)
Comment (YunSuk-Yeo) [reply pkg/blockatlas/tx.go]: ok removed

File: pkg/blockatlas/tx.go (hunk 3 above)
Comment (YunSuk-Yeo) [reply pkg/blockatlas/tx.go]: ok done

File: pkg/blockatlas/tx.go (hunk 1 above)
Comment (vikmeup) [reply pkg/blockatlas/tx.go]: Why two types? you just need one `multi_token_transfer `

File: pkg/blockatlas/tx.go (hunk 3 above)
Comment (vikmeup) [reply pkg/blockatlas/tx.go]: You need to use `blockatlas.Token` as suggested before, if you want amount just wrap `blockatlas.Token` into that object as:
```
struct T {
    token: blockatlas.Token
    amount: Amount
}
```

File: pkg/blockatlas/tx.go [hunk 4]
```diff
@@ -147,6 +149,22 @@ type ContractCall struct {
...
+// MultiCurrencyTransfer describes the transfer of multiple currency native to the platform
+// Example: Cosmos, Terra
+type MultiCurrencyTransfer struct {
+	Currencies []NativeTokenTransfer `json:"currencies"`
+	Fees       []NativeTokenTransfer `json:"fees"`
+}
+
+// MultiCurrencyAnyAction describes any action for multiple currency
+// Example: Cosmos, Terra
+type MultiCurrencyAnyAction struct {
```
Comment (vikmeup) [inline pkg/blockatlas/tx.go]: remove `MultiCurrencyAnyAction `

File: pkg/blockatlas/tx.go [hunk 5]
```diff
@@ -147,6 +149,22 @@ type ContractCall struct {
 	Value string `json:"value"`
 }
 
+// MultiCurrencyTransfer describes the transfer of multiple currency native to the platform
+// Example: Cosmos, Terra
```
Comment (vikmeup) [inline pkg/blockatlas/tx.go]: remove `// Example: Cosmos, Terra`

File: pkg/blockatlas/tx.go [hunk 6]
```diff
@@ -147,6 +149,22 @@ type ContractCall struct {
...
+type MultiCurrencyTransfer struct {
+	Currencies []NativeTokenTransfer `json:"currencies"`
+	Fees       []NativeTokenTransfer `json:"fees"`
+}
+
+// MultiCurrencyAnyAction describes any action for multiple currency
+// Example: Cosmos, Terra
```
Comment (vikmeup) [inline pkg/blockatlas/tx.go]: remove `// Example: Cosmos, Terra`

File: pkg/blockatlas/tx.go (hunk 3 above)
Comment (YunSuk-Yeo) [reply pkg/blockatlas/tx.go]: ok

File: pkg/blockatlas/tx.go (hunk 1 above)
Comment (YunSuk-Yeo) [reply pkg/blockatlas/tx.go]: For the staking operations, fee and reward can also be multiple currencies.

File: pkg/blockatlas/tx.go (hunk 6 above)
Comment (YunSuk-Yeo) [reply pkg/blockatlas/tx.go]: done

File: pkg/blockatlas/tx.go (hunk 5 above)
Comment (YunSuk-Yeo) [reply pkg/blockatlas/tx.go]: done

**Comment by vikmeup** [comment]
Remove `MultiCurrencyAnyAction`

**Comment by YunSuk-Yeo** [comment]
> Remove `MultiCurrencyAnyAction`

Done, but later the staking operations on multi-currency chain will require new tx type to describe withdrawn rewards & fees.

If new chain is connected to cosmoshub and the fee is paid with new denom, the withdrawn rewards can be multiple currency.

**Pull request merged**